controller/manager/controllermanager.go

```diff
@@ -315,16 +372,133 @@ func (c *ControllerManager) addControllerGroup(namespace string) error {
 	return nil
 }
 
+// clean removes and controller groups that no longer correspond to a valid namespace within
+// the Kubernetes cluster, e.g. in the event that a namespace has been deleted.
+func (c *ControllerManager) clean(namespace string) error {
+
+	if !c.sem.TryAcquire(1) {
+		return fmt.Errorf("controller group clean already in progress, namespace %s will not "+
+			"clean", namespace)
+	}
+	defer c.sem.Release(1)
+
+	log.Debugf("Controller Manager: namespace %s acquired clean lock and will clean the "+
+		"controller groups", namespace)
+
+	nsList, err := ns.GetCurrentNamespaceList(c.controllers[namespace].kubeClientset,
+		c.installationName, c.namespaceOperatingMode)
+	if err != nil {
+		log.Errorf(err.Error())
+	}
+
+	for controlledNamespace := range c.controllers {
+		cleanNamespace := true
+		for _, currNamespace := range nsList {
+			if controlledNamespace == currNamespace {
+				cleanNamespace = false
+				break
+			}
+		}
+		if cleanNamespace {
+			log.Debugf("Controller Manager: removing controller group for namespace %s",
+				controlledNamespace)
+			c.removeControllerGroup(controlledNamespace)
+		}
+	}
+
+	return nil
+}
+
+// hasListerPrivs verifies the Operator has the privileges required to start the controllers
+// for the namespace specified.
+func (c *ControllerManager) hasListerPrivs(namespace string) bool {
+
+	controllerGroup := c.controllers[namespace]
+
+	var err error
+	var hasCrunchyPrivs, hasCorePrivs, hasBatchPrivs bool
+
+	for _, listerResource := range listerResourcesCrunchy {
+		hasCrunchyPrivs, err = ns.CheckAccessPrivs(controllerGroup.kubeClientset,
+			map[string][]string{listerResource: []string{"list"}},
+			crv1.GroupName, namespace)
+		if err != nil {
+			log.Errorf(err.Error())
+		} else if !hasCrunchyPrivs {
+			log.Errorf("Controller Manager: Controller Group for namespace %s does not have the "+
+				"required list privileges for resource %s in the Crunchy API",
```

Comment:
The proper name would be `crunchydata.com` API if we're going for that.

Reply:
Or `crunchydata` per the `vendor` label

Reply:
Replaced with the `GroupName` for our API, i.e. `crunchydata.com`.